### PR TITLE
refactor: remove loops in cli/tests/unit/http_test.ts

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/13988

### CODE REVIEW

File: cli/tests/unit/http_test.ts [hunk 1]
```diff
@@ -1866,31 +1872,30 @@ Deno.test("upgradeHttp unix", {
...
-    for await (const conn of listener) {
-      const httpConn = Deno.serveHttp(conn);
-      const maybeReq = await httpConn.nextRequest();
-      assert(maybeReq);
-      const { request, respondWith } = maybeReq;
-      const p = Deno.upgradeHttp(request);
+    const conn = await listener.accept();
+    listener.close();
+    const httpConn = Deno.serveHttp(conn);
+    const maybeReq = await httpConn.nextRequest();
```
Comment (aapoalas) [inline cli/tests/unit/http_test.ts:1878]: Perhaps `reqEvent` like all other tests?

File: cli/tests/unit/http_test.ts [hunk 2]
```diff
@@ -47,15 +47,18 @@ async function writeRequestAndReadResponse(conn: Deno.Conn): Promise<string> {
...
+    const reqEvent = await httpConn.nextRequest();
+    assert(reqEvent);
+    const { request, respondWith } = reqEvent;
+    assertEquals(new URL(request.url).href, "http://127.0.0.1:4501/");
+    assertEquals(await request.text(), "");
+    await respondWith(
+      new Response("Hello World", { headers: { "foo": "bar" } }),
+    );
+    httpConn.close();
+    listener.close();
```
Comment (aapoalas) [inline cli/tests/unit/http_test.ts:61]: This is the only test case where the listener is closed last, all others close it right after receiving `conn`. Consider unifying this?

**Comment by aapoalas** [comment]
LGTM

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**